### DISCUSSION TITLE
Front Camera

### DISCUSSION THREAD

**Discussion by eyeisystems** [tesla-android/issue-tracker#276] [Ideas]
Seeing as though new Tesla 3 / Y come with no USS, i'm looking at getting a number plate camera / sensor - something like this to be used on the front of the car: 

https://www.dolphinparkingsensors.co.uk/product/dolphin-automotive-pps200c-camera-numberplate

I'm thinking that it would be great to integrate the compostie feed from the camera into tesla android, so that you could view the camera live when forwards parking. 

We all know that Tesla Vision is terrible so this at least appears to be a reasonable interim solution :)

**Comment by mikegapinski** [comment]
It will work with a USB grabber that is available for Android tablets/headunits. Getting it to pop up automatically when you reverse is another story. I am not doing integrated CAN hardware anytime soon due to security concerns, but it is doable with any bluetooth module you would normally use for ScanMyTesla.

You can get the video cable through near the MCU on the right side of the cabin without making any holes. I'd take the glovebox out for better access(6 screws) but it should not be hard. It is a very easy car to work with, all the clips are standard and easily obtainable if you break one.

**Reply by mikegapinski** [reply]
Damn, this kit even has a buzzer. Nice! there are tons of harnesses near the MCU to hide it nicely :)

**Reply by mikegapinski** [reply]
And... Given that it has a buzzer it should not be all that hard to get a gpio pin to register that you are approaching something and launch the camera app in Tesla Android. 

If you get it to work on Raspbian(WiringPi) I can potentially port it to Android and integrate it with the platform for launching intents.

**Reply by eyeisystems** [reply]
I'm initally looking at using it for the front given there is already a camera for the back on reverse already :) I'll look at WiringPi and GPIO on the buzzer to fire up the camera on approach. 

They also do a model without the camera which i can potentially use for the reverse to give sensors although probably just as easy wiring it to the reverse light to enable power to activate the buzzer and not integrate them with TeslaAndroid. i imagine most users if they wanted an additional camera on reverse with integrated sensors could use the same method as i'm using on the front anyway.

I'll keep you posted :)

**Reply by mikegapinski** [reply]
Happy to help. When looking for the USB grabber get the one with Linux USB Video Class (UVC) support, it behaves like a webcam and works without any extra drivers. This app is known to work: https://play.google.com/store/apps/details?id=com.shenyaocn.android.usbcamerapro&hl=pl&gl=US 

The buzzer integration looks like this if you were to connect it to the pi. Your circuit would work in reverse; 
![](https://i.stack.imgur.com/sWhLp.png) It all depends on the voltage used by the buzzer. A proof of concept on a breadboard that prints something in the console when you wave your hand near the uss would be enough to start on my end

**Reply by eyeisystems** [reply]
Hopefully it should arrive tomorrow so will get started on it this weekend :)

**Reply by dandelionclock** [reply]
This is really interesting. How are you going to power the camera? I assume it is 12V but Tesla’s low voltage battery is 16V? I’m looking for a solution as well although I don’t care too much about the buzzer part. please keep us posted with your project.

**Comment by miaszh** [comment]
did you get it to work?

looks very interesting!

**Comment by rege245** [comment]
i am also interested. what is the progress?
